tests/test-backend-ops.cpp

```diff
@@ -56,7 +56,18 @@ static void init_tensor_uniform(ggml_tensor * tensor, float min = -1.0f, float m
         GGML_ASSERT(size % ggml_blck_size(tensor->type) == 0);
         std::vector<uint8_t> dataq(ggml_row_size(tensor->type, size));
         int64_t hist[16];
-        ggml_quantize_chunk(tensor->type, data.data(), dataq.data(), 0, size/tensor->ne[0], tensor->ne[0], hist, nullptr);
+        std::vector<float> imatrix(tensor->ne[0], 1.0f); // dummy importance matrix
+        const float * im = imatrix.data();
+        if (!ggml_quantize_requires_imatrix(tensor->type)) {
+            // when the imatrix is optional, we want to test both quantization with and without imatrix
+            std::random_device rd;
+            std::default_random_engine generator(rd());
+            std::uniform_int_distribution<int> distribution(0, 1);
+            if (distribution(generator)) {
+                im = nullptr;
+            }
```

Comment:
I know this code isn't very important, but creating a new random device and then a new random generator every time you want a random boolean is definitely an anti-pattern. You should at least make `generator` static.

Reply:
btw `init_thread` is doing the same thing.

Reply:
For `init_thread` I'm not sure how to improve - would `static thread_local` achieve anything? Simply `static` is a data race AFAIK

Either way - it's OK as it is for this part of the code

Reply:
If it were thread-local you would have to initialize it separately, which would be ugly. If you wanted to improve it you could initialize the random generators at the top of init_tensor_uniform in advance like this (`static` so it happens only once):
```cpp
static std::vector<std::default_random_engine> generators = [n_threads]() {
    std::vector<std::default_random_engine> vec;
    vec.reserve(n_threads);
    std::random_device rd;
    for (size_t i = 0; i < n_threads; i++) { vec.push_back(rd()); }
    return vec;
}();
```

And then pass them to init_thread by reference.